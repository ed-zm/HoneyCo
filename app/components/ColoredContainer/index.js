/**
 *
 * ColoredContainer
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { Container } from 'reactstrap'
import styled from 'styled-components'
import HoneyCoDivider from '../HoneyCoDivider/index'

const StyledContainer = styled(Container)`
  background-color: ${(props) => props.backgroundColor};
  color: ${(props) => props.textColor};
  padding-top: ${(props) => props.title || !props.divider ? '15px' : '1px'};
  padding-bottom: 15px;
`

const PrettyHeader = styled.h1`
  text-align: center;
`

const ColoredContainer = (props) => {
  const { backgroundColor, textColor, title, divider, children } = props
  return (
    <StyledContainer fluid backgroundColor={backgroundColor} textColor={textColor} title={title} divider={divider}>
      {title ? <PrettyHeader>{title}</PrettyHeader> : ''}
      {divider ? <HoneyCoDivider color={textColor} backgroundColor={backgroundColor} /> : ''}
      {children}
    </StyledContainer>
  )
}

ColoredContainer.propTypes = {
  backgroundColor: PropTypes.string.isRequired,
  textColor: PropTypes.string.isRequired,
  children: PropTypes.node,
  title: PropTypes.string,
  divider: PropTypes.bool,
}

ColoredContainer.defaultProps = {
  children: null,
  title: '',
  divider: true,
}

export default ColoredContainer
