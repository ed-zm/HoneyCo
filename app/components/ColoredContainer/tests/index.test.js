import React from 'react'
import { shallow } from 'enzyme'
import ColoredContainer from '../index'

describe('<ColoredContainer />', () => {
  it('It should render the ColoredContainer Component', () => {
    const wrapper = shallow(<ColoredContainer/>)
    expect(wrapper).toMatchSnapshot()
  })
})
