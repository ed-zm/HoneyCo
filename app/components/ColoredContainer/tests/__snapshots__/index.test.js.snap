// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColoredContainer /> It should render the ColoredContainer Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(gt)
    backgroundColor={undefined}
    divider={true}
    fluid={true}
    textColor={undefined}
    title=""
>
    
    <HoneyCoDivider
        backgroundColor={undefined}
        color={undefined}
    />
</Styled(gt)>,
  "nodes": Array [
    <Styled(gt)
      backgroundColor={undefined}
      divider={true}
      fluid={true}
      textColor={undefined}
      title=""
>
      
      <HoneyCoDivider
            backgroundColor={undefined}
            color={undefined}
      />
</Styled(gt)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ColoredContainer
        divider={true}
        title=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": null,
          "divider": true,
          "title": "",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(gt)
          backgroundColor={undefined}
          divider={true}
          fluid={true}
          textColor={undefined}
          title=""
>
          
          <HoneyCoDivider
                    backgroundColor={undefined}
                    color={undefined}
          />
</Styled(gt)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(gt)
          backgroundColor={undefined}
          divider={true}
          fluid={true}
          textColor={undefined}
          title=""
>
          
          <HoneyCoDivider
                    backgroundColor={undefined}
                    color={undefined}
          />
</Styled(gt)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ColoredContainer
    divider={true}
    title=""
/>,
}
`;
